### PR TITLE
docs: improve search performance

### PR DESCRIPTION
**Pull request by soonoo**
Related issue: #1383 

What I did:
- renders only part of the search results and render the rest when scroll reaches to the end.
- debounces rendering of search results.

elasticlunr's search function seems fast enough and what slows website was search component's `render` function. (About 50 ~ 300 items are rendered whenever user types)

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -17,9 +17,11 @@
     "gatsby-source-filesystem": "^2.0.28",
     "gatsby-transformer-remark": "^2.3.8",
     "lodash": "^4.17.11",
+    "lodash.debounce": "^4.0.8",
```
Comment (zeke) [inline docs/package.json]: looks like the full `lodash` library is already in the mix, so you probably don't need `lodash.debounce`

File: docs/src/components/search.js [hunk 2]
```diff
@@ -85,15 +97,32 @@ export default class Search extends Component {
...
+      // Map over each ID and return the full document
+      .map(({ ref }) => this.index.documentStore.getDoc(ref))
+
+    let height = 0
+    let visibleResultsCount
+    // Calculate maximum count of results that fits in current viewport height
+    for(const [index, doc] of results.entries()) {
+      // 54, 26 are the height of search results item.
+      // They are unknonw until rendered. Is there a better way to set these values?
+      if(doc.type === 'API method') height += 54
```
Comment (zeke) [inline docs/src/components/search.js]: There probably is a more dynamic way to do this, but if you end up keeping this implementation I would suggest changing these [magic numbers](https://en.wikipedia.org/wiki/Magic_number_%28programming%29) into `const` declarations.

File: docs/package.json (hunk 1 above)
Comment (soonoo) [reply docs/package.json]: I didn't konw that, uninstalled lodash.debounce!

File: docs/src/components/search.js (hunk 2 above)
Comment (soonoo) [reply docs/src/components/search.js]: I think it gets a bit complicated with dynamic way so I just end up keeping numeric values. Please check commits below.

File: docs/package.json (hunk 1 above)
Comment (gr2m) [reply docs/package.json]: Oops sorry I missed that bit, I think we need the `lodash.debounce` separately. `lodash` is used only in built scripts, not in the frontend of the documentation. The `lodash.debounce` method is currently imported in `docs/src/components/search.js` so we have to keep it, even though it looks weird in package.json

Comment (soonoo) [reply docs/package.json]: https://github.com/octokit/rest.js/blob/b8f2097ef7bf03ffde602a6ef866898ee025dbf3/docs/src/components/search.js#L2

I forgot to change import statement and pushed the changed code. Is this okay with just `lodash` dependency?

Comment (gr2m) [reply docs/package.json]: I’m not sure if gatsby does proper tree shaking to remove unused code, but it should be okay. Maybe compare the download size of JS assets locally to what we currently have in production? I thenk you have to run `gatsby build` and then `gatsby serve`. I don’t think `gatsby develop` will do all the bundling and minification

Comment (soonoo) [reply docs/package.json]: Above import statement imports whole `lodash` library so changed it to import only a debounce function.
https://github.com/octokit/rest.js/blob/2bb9a17bd380bf0eaf9f1bf9baeb7d40d9bf357d/docs/src/components/search.js#L2

without debounce (24.21KB transferred and file size is 74.02KB):
2019-06-04T15:57:49

with debouce:
2019-06-04T15:57:40

Is this affordable?

Comment (gr2m) [reply docs/package.json]: There is probably use for optimization but that can be addressed in a future PR :)

**Review by gr2m — APPROVED** [review]
This is great, thanks a lot Soonwoo! The search feels _much_ better now!

**Pull request merged**